File: basics/01-array/0109-167.go
func twoSum(numbers []int, target int) []int {
    if len(numbers) == 2 {
        return []int{1, 2}
    } 

    left := 0
    right := len(numbers)-1
    
    for {
        sum := numbers[left] + numbers[right]
        if (target == sum) {
            return []int{left+1, right+1}
        } else if (target < sum) {
            right -= 1
        } else {
            left += 1
        }
    }
}

File: basics/01-array/0109-167.cpp
/*
 * leetcode 167
 * two sum II
 *
 */

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int left = 0, right = numbers.size()-1;

        while (left < right) {
            int s = numbers[left] + numbers[right];
            if (s == target) {
                break;
            } else if (s > target){
                right--;
            } else {
                left++;
            }
        }
        return {left+1, right+1};
    }
};

